autodispose-android/src/androidTest/java/com/uber/autodispose/android/AutoDisposeViewHolderViewTest.java

```diff
@@ -0,0 +1,146 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.android;
+
+import android.app.Instrumentation;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import com.uber.autodispose.ObservableScoper;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Action;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.swipeUp;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static com.google.common.truth.Truth.assertThat;
+
+/**
+ * Bit of an odd one, but this test works by flinging the list and ensuring the number of recycles
+ * lines up with the number of disposals. We can't monitor recycling + immediate disposable directly
+ *  because it's actually async.
+ */
+@RunWith(AndroidJUnit4.class) public final class AutoDisposeViewHolderViewTest {
+  @Rule public final ActivityTestRule<AutoDisposeViewHolderTestActivity> activityRule =
+      new ActivityTestRule<>(AutoDisposeViewHolderTestActivity.class);
+
+  private final Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
+  private final AtomicInteger recycledCounter = new AtomicInteger();
+  private final AtomicInteger disposeCounter = new AtomicInteger();
+
+  private RecyclerView view;
+  private ViewInteraction interaction;
+  private ViewDirtyIdlingResource viewDirtyIdler;
+
+  @Before public void setUp() {
+    AutoDisposeViewHolderTestActivity activity = activityRule.getActivity();
+    view = activity.recyclerView;
+    interaction = onView(withId(android.R.id.primary));
+    viewDirtyIdler = new ViewDirtyIdlingResource(activity);
+    Espresso.registerIdlingResources(viewDirtyIdler);
+  }
+
+  @After public void tearDown() {
+    Espresso.unregisterIdlingResources(viewDirtyIdler);
+  }
+
+  @Test public void basicTest() {
```

Comment:
can you also add a test for a fling to bottom, then fling back to top and check the count? this would ensure that behavior is correct when a viewholder is rebound